ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -248,6 +248,22 @@ internal constructor(
     }
   }
 
+  fun saveCurrentState() {
+    getTaskViewModel(currentTaskId.value)?.let {
+      if (!data.containsKey(it.task)) {
+        val validationError = it.validate()
+        if (validationError != null) {
+          clearDraft()
```

Comment:
Do we really want to clear the draft (i.e. all tasks) even if the current one is failing validation checks? IMO, we should just return and not clear everything. wdyt?

Reply:
the reason to clear is, when we move back to home screen, there is a check which opens the draft collections, so need to clear

did you try removing this and see how app behave?

Reply:
fixed

Reply:
Consider a scenario where you have 5 tasks and 3 of them are already filled. On 4th task, the user decided to kill the app abruptly. If we don't clear the draft, then atleast 3 tasks would be revived.